### PR TITLE
RR-335 CIAG Induction created/updated message specs

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR adds the message specifications for CIAG induction events - specifically whenever a CIAG induction is created or updated.

### CODE REVIEW

File: spec/schemas/ciag-induction/ciag-induction-created.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: ciag-induction.created
+title: CIAG Induction Created
+summary: A CIAG induction has been created for a Prisoner.
+contentType: application/json
+payload:
+  type: object
+  properties:
+    eventType:
+      type: string
+      example: "ciag-induction.created"
```
Comment (nathanrussell-moj-digital) [inline spec/schemas/ciag-induction/ciag-induction-created.yml:10]: Given that this is the schema for one specific event type, the example in this case feels like less of an example, and more of the 1 and only 1 value that is supported here?
Could/should this be something like:
```
properties:
  eventType:
    type: string
    enum:
      - "ciag-induction.created"
```

Would that work?

File: spec/schemas/ciag-induction/ciag-induction-created.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    version:
+      type: integer
+      example: 1
+    detailUrl:
+      type: string
+      example: "https://learning-and-work-progress.hmpps.service.justice.gov.uk/plan/A0670DZ/view/overview"
+    occurredAt:
+      type: string
+      example: "2023-10-05T12:00:00.000Z"
```
Comment (nathanrussell-moj-digital) [inline spec/schemas/ciag-induction/ciag-induction-created.yml:19]: Is it worth adding `format: date-time` to this property? (`occurredAt`)

File: spec/schemas/ciag-induction/ciag-induction-created.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    description:
+      type: string
+      example: "CIAG Induction Completed"
```
Comment (nathanrussell-moj-digital) [inline spec/schemas/ciag-induction/ciag-induction-created.yml:22]: Similar to the comment about `eventType` ... is this an example, or is it the 1 and only 1 description value that we would expect producers of these messages to provide?

I guess the question is what is the purpose of the `description` field? What are we as consumers of the message going to do with it?

Does it matter if the CIAG API produces a message with a description field of `CIAG Induction Completed` or `A CIAG Induction was completed for prisoner A1234BC` or `wibble-wobble` ?

File: spec/schemas/ciag-induction/ciag-induction-created.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          type: array
+          items:
+            type: object
+            properties:
+              type:
+                type: string
+                example: "CRN"
+              value:
+                type: string
+                example: "D001022"
```
Comment (nathanrussell-moj-digital) [inline spec/schemas/ciag-induction/ciag-induction-created.yml:57]: Do we need (want?) a `required` list for the top level objects? Like there is in the "parent" schema?

File: spec/schemas/ciag-induction/ciag-induction-created.yml (hunk 1 above)
Comment (mattwills-moj-digital) [reply spec/schemas/ciag-induction/ciag-induction-created.yml]: Good idea. Done.

File: spec/schemas/ciag-induction/ciag-induction-created.yml (hunk 3 above)
Comment (mattwills-moj-digital) [reply spec/schemas/ciag-induction/ciag-induction-created.yml:24]: I'm not sure what it's used for exactly, but it's in every other message spec in this repo! I'd rather not enforce anything on it.

File: spec/schemas/ciag-induction/ciag-induction-created.yml (hunk 2 above)
Comment (mattwills-moj-digital) [reply spec/schemas/ciag-induction/ciag-induction-created.yml:21]: Good point, done.

**Comment by mattwills-moj-digital** [comment]
Closing as going to follow a common approach for DPS projects.

**Pull request closed without merging**